pulsar-transformations/src/main/resources/config-schema.yaml

```diff
@@ -141,6 +142,57 @@ components:
               description: The delimiter to use when concatenating the field names.
               default: _
         - "$ref": "#/components/schemas/Part"
+    ComputeFields:
+      allOf:
+        - "$ref": "#/components/schemas/Step"
+        - type: object
+          description: Adds or updates fields dynamically based on an expression (Currently only AVRO is supported).
+          properties:
+            type:
+              type: string
+              enum:
+                - compute-fields
+            fields:
+              type: [ array, 'null' ]
+              items:
+                type:
+                  - object
+                minLength: 1
+                properties:
+                  name:
+                    type: string
+                    description: The name of the field to compute. If the field name already exist, it will be overwritten.
+                  expression:
+                    type: string
+                    description: An EL-like expression to compute the field value. It can refer to other attributes in the record.
+                  type:
+                    type: string
+                    description: The data type of the computed fields. It should match with the output of the expression.
+                    enum:
+                      - STRING
+                      - INT32
+                      - INT64
+                      - FLOAT
+                      - DOUBLE
+                      - BOOLEAN
+                  part:
+                    type: string
+                    description: The part of the message to apply the computed fields to. If left empty, the fields will apply to the `value`.
+                    enum:
+                      - key
+                      - value
+                    default: value
+                description: The list of fields to compute
+                message:
+                  minLength: "field name in 'fields' must not be empty"
+                required:
+                  - name
+                  - expression
+                  - type
```

Comment:
Is type required or could it be inferred ?

Reply:
In the current implementation it is explicit. I think inference will open the door for some issues and would prefer predictable from the user on the schema. For example if I user types `${1}` and we bind all variables to `Object.class`:
```
ValueExpression e = factory.createValueExpression(context, "${1}", Object.class); 
System.out.println(e.getValue(context).getClass().getName()); 

OUTPUT:
java.lang.Long
```

You'll get a long, but int would be a better fit. We an alway do our own inference like to check if this long can be stored in int and use int, but it can get tricky with other types. Also I'm thinking, if one day we switch the underlying JSTL engine, it is hard to guarantee it'll return the same types...